# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1749505624 -7200
# Parent  d9d118a3af71727f6c7f668fd3d3e051684d0e95
Misc: Manually sort folders: Also read previous attribute "folderSortOrder" for compatibility with 128 implementation.

diff --git a/mailnews/base/src/nsMsgDBFolder.cpp b/mailnews/base/src/nsMsgDBFolder.cpp
--- a/mailnews/base/src/nsMsgDBFolder.cpp
+++ b/mailnews/base/src/nsMsgDBFolder.cpp
@@ -1195,9 +1195,13 @@ NS_IMETHODIMP nsMsgDBFolder::ReadFromFol
     nsIMsgFolderCacheElement* element) {
   nsresult rv = NS_OK;
 
   element->GetCachedUInt32("flags", &mFlags);
-  element->GetCachedUInt32("userSortOrder", &mUserSortOrder);
+  if (NS_FAILED(element->GetCachedUInt32("userSortOrder", &mUserSortOrder)) ||
+      mUserSortOrder == (uint32_t)nsIMsgFolder::NO_SORT_VALUE) {
+    // Try getting the pre-140 property.
+    element->GetCachedUInt32("folderSortOrder", &mUserSortOrder);
+  }
   element->GetCachedInt32("totalMsgs", &mNumTotalMessages);
   element->GetCachedInt32("totalUnreadMsgs", &mNumUnreadMessages);
   element->GetCachedInt32("pendingUnreadMsgs", &mNumPendingUnreadMessages);
   element->GetCachedInt32("pendingMsgs", &mNumPendingTotalMessages);
@@ -1275,8 +1279,10 @@ NS_IMETHODIMP nsMsgDBFolder::WriteToFold
   nsresult rv = NS_OK;
 
   element->SetCachedUInt32("flags", mFlags);
   element->SetCachedUInt32("userSortOrder", mUserSortOrder);
+  // Write the pre-140 property as well.
+  element->SetCachedUInt32("folderSortOrder", mUserSortOrder);
   element->SetCachedInt32("totalMsgs", mNumTotalMessages);
   element->SetCachedInt32("totalUnreadMsgs", mNumUnreadMessages);
   element->SetCachedInt32("pendingUnreadMsgs", mNumPendingUnreadMessages);
   element->SetCachedInt32("pendingMsgs", mNumPendingTotalMessages);
diff --git a/mailnews/db/msgdb/src/nsDBFolderInfo.cpp b/mailnews/db/msgdb/src/nsDBFolderInfo.cpp
--- a/mailnews/db/msgdb/src/nsDBFolderInfo.cpp
+++ b/mailnews/db/msgdb/src/nsDBFolderInfo.cpp
@@ -286,14 +286,23 @@ NS_IMETHODIMP nsDBFolderInfo::SetFolderD
 }
 
 NS_IMETHODIMP nsDBFolderInfo::GetUserSortOrder(uint32_t* userSortOrder) {
   NS_ENSURE_ARG_POINTER(userSortOrder);
-  return GetUint32Property("userSortOrder", nsIMsgFolder::NO_SORT_VALUE,
-                           userSortOrder);
+  nsresult rv = GetUint32Property("userSortOrder", nsIMsgFolder::NO_SORT_VALUE,
+                                  userSortOrder);
+  if (NS_FAILED(rv) || *userSortOrder == (uint32_t)nsIMsgFolder::NO_SORT_VALUE) {
+    // Try getting the pre-140 property.
+    return GetUint32Property("folderSortOrder", nsIMsgFolder::NO_SORT_VALUE,
+                             userSortOrder);
+  }
+  return NS_OK;
 }
 
 NS_IMETHODIMP nsDBFolderInfo::SetUserSortOrder(uint32_t userSortOrder) {
-  return SetUint32Property("userSortOrder", userSortOrder);
+  nsresult rv = SetUint32Property("userSortOrder", userSortOrder);
+  NS_ENSURE_SUCCESS(rv, rv);
+  // Write the pre-140 property as well.
+  return SetUint32Property("folderSortOrder", userSortOrder);
 }
 
 NS_IMETHODIMP nsDBFolderInfo::GetHighWater(nsMsgKey* result) {
   // Sanity check highwater - if it gets too big, other code
